### PR TITLE
DOC: Improving the docstring of Series.str.upper and related

### PR DESCRIPTION
**Pull request by Hamishpk**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [X] PR title is "DOC: update the <your-function-or-method> docstring"
- [X] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [X] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [X] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
################################################################################
##################### Docstring (pandas.Series.str.upper)  #####################
################################################################################

Convert strings in the Series/Index to uppercase.

Equivalent to :meth:`str.upper`.

Returns
-------
Series/Index of objects

See Also
--------
Series.str.lower : Converts all characters to lower case.
Series.str.upper : Converts all characters to upper case.
Series.str.title : Converts first character of each word to upper case and
    remaining to lower case.
Series.str.capitalize : Converts first character to upper case and
    remaining to lower case.
Series.str.swapcase : Converts upper case to lower case and lower case to
    upper case.

Examples
--------
>>> s = pd.Series(['lower', 'CAPITALS', 'this is a sentence', 'SwApCaSe'])
>>> s.str.lower()
0                 lower
1              capitals
2    this is a sentence
3              swapcase
dtype: object

>>> s.str.upper()
0                 LOWER
1              CAPITALS
2    THIS IS A SENTENCE
3              SWAPCASE
dtype: object

>>> s.str.title()
0                 Lower
1              Capitals
2    This Is A Sentence
3              Swapcase
dtype: object

>>> s.str.capitalize()
0                 Lower
1              Capitals
2    This is a sentence
3              Swapcase
dtype: object

>>> s.str.swapcase()
0                 LOWER
1              capitals
2    THIS IS A SENTENCE
3              sWaPcAsE
dtype: object

################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.Series.str.upper" correct. :)

```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.

Checklist for other PRs (remove this part if you are doing a PR for the pandas documentation sprint):

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/strings.py [hunk 1]
```diff
@@ -2145,11 +2145,61 @@ def rindex(self, sub, start=0, end=None):
...
+        remaining to lower case.
+    Series.str.capitalize : Converts first character to upper case and
+        remaining to lower case.
+    Series.str.swapcase : Converts upper case to lower case and lower case to
+        upper case.
+
+    Examples
+    --------
+    >>> s = pd.Series(['lower', 'CAPITALS', 'this is a sentence', 'SwApCaSe'])
+    >>> s.str.lower()
```
Comment (datapythonista) [inline pandas/core/strings.py]: Could you show the original `s`, (simply with a line `>>> s` and its output). And then leave a blank line between it and the `lower` method, so the lower doesn't get grouped with the constructor and the original, but in a separate block as the rest?

Besides that, lgtm.

File: pandas/core/strings.py [hunk 2]
```diff
@@ -2145,11 +2145,61 @@ def rindex(self, sub, start=0, end=None):
...
     Equivalent to :meth:`str.%(method)s`.
 
     Returns
     -------
-    converted : Series/Index of objects
+    Series/Index of objects
+
+    See Also
+    --------
+    Series.str.lower : Converts all characters to lower case.
```
Comment (jorisvandenbossche) [inline pandas/core/strings.py]: "lower case" -> "lowercase" ? 
Or at least we should be consistent with what is substituted in the summary line (currently this is "lowercase")

**Review by datapythonista — APPROVED** [review]
lgtm

**Comment by TomAugspurger** [comment]
Fixed the doctest.

thanks @Hamishpk

**Pull request merged**